src/Transformers/ModelTransformer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Spatie\LaravelTypeScriptTransformer\Transformers;
+
+use Spatie\TypeScriptTransformer\Structures\TransformedType;
+
+use DB;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Schema;
+use ReflectionClass;
+
+/**
+ * A Transformer that generates Typescript definitions for a Model by inspecting the DB Schema
+ */
+class ModelTransformer implements Transformer
+{
+    public function transform(ReflectionClass $class, string $name): ?TransformedType
+    {
+        if (!is_subclass_of($class->name, Model::class)) {
+            return null;
+        }
+
+        /** @var Model $modelInstance */
+        $modelInstance = $class->newInstanceWithoutConstructor();
+
+        $table = $modelInstance->getTable();
+        $columns = Schema::getColumnListing($table);
+
+        $hidden = $modelInstance->getHidden();
+
+        $serializedColumns = array_diff($columns, $hidden);
```

Comment:
Here is where I believe there should also be a references to `$modelInstance->getVisible()`. However, I'm not yet sure how `$hidden` and `$visible` interact when serializing.